### DISCUSSION TITLE
How to bail out from the store update from within the setState?

### DISCUSSION THREAD

**Discussion by denisborovikov** [pmndrs/zustand#1761] [Q&A] [answered]
Does calling the `setState` always cause the store update, or it's possible to bail out returning `undefined` or the previous state? 

```js
useBearStore.setState(({ bears }) => ({
  bears: bears === 0 ? bears + 1 : bears
}))
```
or
```js
useBearStore.setState(({ bears }) => {
  return bears === 0 ? {bears: bears + 1} : undefined
})
```

Thank you!

**Comment by dai-shi** [comment, marked as answer]
To bail out, you can return the previous state (must be referentially identical):
```js
useBearStore.setState((state) => {
  return state.bears === 0 ? {bears: state.bears + 1} : state
})
```

Maybe it's worth for docs.

**Reply by denisborovikov** [reply]
Ah got it. You have to return the entire state, not the slice. Thank you.
